RELEASES.md

```diff
@@ -205,3 +205,95 @@ To create a minor release:
 6. Create a pull request back to master with the CHANGELOG & version changes from the latest release.
    - Remove all `R:minor` labels from the pull requests that were included in the release.
    - Do not merge the backport branch into master.
+
+
+## Minor Release Checklist
+
+The following set of steps are performed on all releases that increment
+the _minor_ version. These steps ensure that Tendermint is well tested, stable,
+and suitable for adoption by the various diverse projects that rely on Tendermint.
+
+### Feature Freeze
+
+Ahead of any minor version release of Tendermint, the software enters a two
+week 'Feature Freeze'. A feature freeze means that _no_ new features are added
+to the code base. The Tendermint team instead directs all attention to ensuring
```

Comment:
> So that anyone working on changes that won't make it to the release is not blocked during this period.

Added this. I think having this period directly follow the created backport branch is a good idea. It signals that we have code that is done being added to.